### PR TITLE
plugins: add make plugin

### PR DESCRIPTION
**Pull request by cmatsuoka**
The make plugin is useful to build make-based parts.

Signed-off-by: Claudio Matsuoka <claudio.matsuoka@canonical.com>

- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: craft_parts/plugins/base.py [hunk 1]
```diff
@@ -60,3 +62,40 @@ def out_of_source_build(self) -> bool:
...
+) -> Dict[str, Any]:
+    """Obtain plugin-specifc entries from part properties.
+
+    :param data: A dictionary containing all part properties.
+    :plugin_name: The name of the plugin.
+
+    :return: A dictionary with plugin properties.
+    """
+    plugin_data: Dict[str, Any] = {}
+    prefix = f"{plugin_name}-"
```
Comment (cjp256) [inline craft_parts/plugins/base.py:95]: So apparently I lied when I said this was true for v2. The python plugin has 'requirements' and 'constraints' that don't follow this model, and perhaps there are others...

File: tests/integration/plugins/test_make.py [hunk 2]
```diff
@@ -0,0 +1 @@
+# TODO: add make plugin integration test after executor lands
```
Comment (cjp256) [inline tests/integration/plugins/test_make.py:1]: craft-providers `Executor` landed first lol

File: craft_parts/plugins/base.py (hunk 1 above)
Comment (cjp256) [reply craft_parts/plugins/base.py:95]: Might be worth just dropping this generalization and make each plugin specify the list explicitly.

**Review by cjp256 — APPROVED** [review]

File: craft_parts/plugins/base.py (hunk 1 above)
Comment (sergiusens) [reply craft_parts/plugins/base.py:95]: the reason is legacy, it is nice to know what is consumed by the plugin and what is consumed by the rest of the machinery

**Review by sergiusens — APPROVED** [review]

File: craft_parts/plugins/base.py (hunk 1 above)
Comment (cmatsuoka) [reply craft_parts/plugins/base.py:95]: I verified current v2 snapcraft plugins, `requirements` and `constraints` in the python plugin are the only properties that violate the name prefix rule. Craft-parts could establish that as a hard rule (and legacy cases could be pre-processed if needed?)

Comment (cjp256) [reply craft_parts/plugins/base.py:95]: Seems fair.  It looks like it's under control of the plugins anyhow so it should be OK even as-is?

**Pull request merged**